src/OpenRiaServices.Client.DomainClients.Http/Framework/Http/BinaryHttpDomainClient.cs

```diff
@@ -193,11 +194,60 @@ async Task<QueryCompletedResult> QueryAsyncCoreContinuation()
         /// <param name="cancellationToken"></param>
         private Task<HttpResponseMessage> PostAsync(string operationName, IDictionary<string, object> parameters, List<ServiceQueryPart> queryOptions, CancellationToken cancellationToken)
         {
-            var request = new HttpRequestMessage(HttpMethod.Post, operationName)
+            var request = new HttpRequestMessage(HttpMethod.Post, operationName);
+
+            using (var ms = new MemoryStream())
+            using (var writer = System.Xml.XmlDictionaryWriter.CreateBinaryWriter(ms, null, null, ownsStream: false))
             {
-                Content = new BinaryXmlContent(this, operationName, parameters, queryOptions),
-            };
+                // Write message
+                var rootNamespace = "http://tempuri.org/";
+                bool hasQueryOptions = queryOptions != null && queryOptions.Count > 0;
+
+                if (hasQueryOptions)
+                {
+                    writer.WriteStartElement("MessageRoot");
+                    writer.WriteStartElement("QueryOptions");
+                    foreach (var queryOption in queryOptions)
+                    {
+                        writer.WriteStartElement("QueryOption");
+                        writer.WriteAttributeString("Name", queryOption.QueryOperator);
+                        writer.WriteAttributeString("Value", queryOption.Expression);
+                        writer.WriteEndElement();
+                    }
+                    writer.WriteEndElement();
```

Comment:
We write two start elements on row 208 and 209 but only one end element here, will this cause a problem?

Reply:
No, it is handled by writeenddocument
The actual xml content is wrapped in MessageRoot only if there are any messageOptions